File: 5week/Solutions/상하좌우.java
import java.util.*; 

public class Main { 
    public static void main(String[] args) { 
        Scanner sc = new Scanner(System.in); 
        // 1. 입력 받기
        int n = sc.nextInt(); sc.nextLine(); 
        String[] plans = sc.nextLine().split(" "); 
        int x = 1, y = 1; 
        
        // 2. LRUD 이동방향 설정
        int[] dx = {0, 0, -1, 1}; 
        int[] dy = {-1, 1, 0, 0}; 
        char[] moveTypes = {'L', 'R', 'U', 'D'}; 
        
        for (int i = 0; i < plans.length; i++) { 
            char plan = plans[i].charAt(0); 
            int nx = -1, ny = -1; 
          
            for (int j = 0; j < 4; j++) { 
                if (plan == moveTypes[j]) { 
                    nx = x + dx[j]; 
                    ny = y + dy[j]; 
                } 
            } 
          
            if (nx < 1 || ny < 1 || nx > n || ny > n) continue;

            x = nx; 
            y = ny; 
        } 
            System.out.println(x + " " + y); 
    }
}

File: 5week/Solutions/상하좌우.c
#include <stdio.h>

int n;
char plans[100];
char str[100];
int x = 1, y = 1;

int dx[4] = {0, 0, -1, 1};
int dy[4] = {-1, 1, 0, 0};
char moveTypes[4] = {'L', 'R', 'U', 'D'};

int main(void) {
	//code
	scanf_s("%d", &n);
    int count = 0, cnt=0;
    scanf("%[^n]s", str);
    // 이동 계획을 하나씩 확인하기
	while (1) {
	    count++;
	    if (str[count] == '\n' || '\0') break;
	    if (str[count] == ' ') continue;
        plans[cnt] = str[count];
        char plan = plans[cnt];
        // 이동 후 좌표 구하기
        int nx = -1, ny = -1;
        for (int j = 0; j < 4; j++) {
            if (plan == moveTypes[j]) {
                nx = x + dx[j];
                ny = y + dy[j];
            }
        }
        // 공간을 벗어나는 경우 무시
        if (nx < 1 || ny < 1 || nx > n || ny > n)
        continue;
            // 이동 수행
            x = nx;
            y = ny;
            cnt++;
    }
    printf("%d %d \n", x, y);
	return 0;
}
